src/schemas/option-group-based-fields.ts

```diff
@@ -0,0 +1,89 @@
+import * as yup from "yup";
+
+const PREFILL_ACTIONID_REGEX = /^[a-zA-Z0-9_-]+$/;
+const PREFILL_PATH_REGEX = /^[a-zA-Z0-9._-]+$/;
+
+export const OPTION_GROUP_BASED_SCHEMA = yup.object().shape({
+    placeholder: yup.string().optional(),
+    dropdownItems: yup
+        .array()
+        .of(
+            yup.object().shape({
+                label: yup.string().optional(),
+                value: yup.string().optional(),
+            })
+        )
+        .test(
+            "min-2-valid-items",
+            "At least 2 items with valid label and value are required.",
+            function (dropdownItems) {
```

Comment:
```suggestion
            (dropdownItems) => {
```

Reply:
I think using non-arrow functions would be more appropriate for test functions since we are accessing `this`.


[Yup official documents](https://github.com/jquense/yup?tab=readme-ov-file#schematestname-string-message-string--function--any-test-function-schema) states: 
_For non arrow functions, the test context is also set as the function `this`. Watch out, if you access it via `this` it won't work in an arrow function._